pkg/ring/ring.go

```diff
@@ -335,20 +338,49 @@ func (r *Ring) GetAll(op Operation) (ReplicationSet, error) {
 	maxUnavailable := r.cfg.ReplicationFactor / 2
 	numRequired -= maxUnavailable
 
-	ingesters := make([]IngesterDesc, 0, len(r.ringDesc.Ingesters))
+	maxUnavailableZones := len(r.ringZones) - maxUnavailable - 1
+	if maxUnavailableZones < 0 {
```

Comment:
Given the comment about you will not need this anymore, but if you will need again in the future please be aware you can just use `util.Max(0, ...)`